src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -754,42 +756,46 @@ public Entity getBaseVehicle() {
     }
 
     @Override
-    public DataTransactionResult addPassenger(Entity entity) {
+    public boolean hasPassenger(Entity entity) {
+        checkNotNull(entity);
+        return entity.getPassengers().contains(this);
+    }
+
+    @Override
+    public boolean addPassenger(Entity entity) {
         checkNotNull(entity);
         if (entity.getPassengers().contains(this)) {
             throw new IllegalArgumentException(String.format("Cannot add entity %s as a passenger of %s, because the former already has the latter as a passenger!", entity, this));
         }
 
-        final ImmutableList.Builder<EntitySnapshot> passengerSnapshotsBuilder = ImmutableList.builder();
-        passengerSnapshotsBuilder.addAll(getPassengers().stream().map(Entity::createSnapshot).collect(Collectors.toList()));
+        return ((net.minecraft.entity.Entity) entity).startRiding((net.minecraft.entity.Entity) (Object) this, true);
+    }
 
-        final DataTransactionResult.Builder builder = DataTransactionResult.builder();
-        if (!((net.minecraft.entity.Entity) entity).startRiding((net.minecraft.entity.Entity) (Object) this, true)) {
-            return builder.result(DataTransactionResult.Type.FAILURE).reject(new ImmutableSpongeListValue<>(Keys.PASSENGERS, ImmutableList.of(entity
-                    .createSnapshot()))).build();
+    @Override
+    public void removePassenger(Entity entity) {
+        checkNotNull(entity);
+        if (entity.getPassengers().contains(this)) {
+            throw new IllegalArgumentException(String.format("Cannot remove entity %s, because it is not a passenger of %s ", entity, this));
```

Comment:
How would you suggest it? I've been fairly unsure about that line, and changed it a few times by suggestion